pilot/pkg/networking/core/v1alpha3/listener_inbound.go

```diff
@@ -0,0 +1,674 @@
+			})
+		case istionetworking.ListenerProtocolTCP:
+			chains = append(chains, &listener.FilterChain{
+				FilterChainMatch: cc.ToFilterChainMatch(opt),
+				Filters:          lb.buildInboundNetworkFilters(cc),
+				TransportSocket:  buildDownstreamTLSTransportSocket(opt.ToTransportSocket(mtls)),
+				Name:             cc.Name(opt.Protocol),
+			})
+		}
+	}
+	return chains
+}
+
+// buildInboundChainConfigs builds all the application chain configs.
+func (lb *ListenerBuilder) buildInboundChainConfigs() []inboundChainConfig {
+	chainsByPort := make(map[uint32]inboundChainConfig)
+	// No user supplied sidecar scope or the user supplied one has no ingress listeners.
+	if !lb.node.SidecarScope.HasIngressListener() {
+		// We will look at all Services that apply to this proxy and build chains for each distinct port.
+		// Note: this does mean that we may have multiple Services applying to the same port, which introduces a conflict
+		for _, i := range lb.node.ServiceInstances {
+			port := ServicePort{
+				Name:       i.ServicePort.Name,
+				Port:       uint32(i.ServicePort.Port),
+				TargetPort: i.Endpoint.EndpointPort,
+				Protocol:   i.ServicePort.Protocol,
+			}
+
+			cc := inboundChainConfig{
+				telemetryMetadata: util.TelemetryMetadata{InstanceHostname: i.Service.Hostname},
+				port:              port,
+				clusterName:       model.BuildInboundSubsetKey(int(port.TargetPort)),
+				bind:              "0.0.0.0", // TODO ipv6
+				bindToPort:        getBindToPort(networking.CaptureMode_DEFAULT, lb.node),
+			}
+			if i.Service.Attributes.ServiceRegistry == provider.Kubernetes {
+				cc.telemetryMetadata.KubernetesServiceNamespace = i.Service.Attributes.Namespace
+				cc.telemetryMetadata.KubernetesServiceName = i.Service.Attributes.Name
+			}
+			// First, make sure there is a distinct instance used per port.
+			// The Service is *almost* not relevant, but some Telemetry is per-service.
+			// If there is a conflict, we will use the oldest Service. This impacts the protocol used as well.
+			if old, f := chainsByPort[port.TargetPort]; f {
+				reportInboundConflict(lb, old, cc)
+				continue
+			}
+			chainsByPort[port.TargetPort] = cc
+		}
+	} else {
+		for _, i := range lb.node.SidecarScope.Sidecar.Ingress {
+			port := ServicePort{
+				Name:       i.Port.Name,
+				Port:       i.Port.Number,
+				TargetPort: i.Port.Number, // No targetPort support in the API
+				Protocol:   protocol.Parse(i.Port.Protocol),
+			}
+			bindtoPort := getBindToPort(i.CaptureMode, lb.node)
+			// Skip ports we cannot bind to
+			if !lb.node.CanBindToPort(bindtoPort, port.TargetPort) {
+				log.Warnf("buildInboundListeners: skipping privileged sidecar port %d for node %s as it is an unprivileged proxy",
+					i.Port.Number, lb.node.ID)
+				continue
+			}
+
+			cc := inboundChainConfig{
+				// Sidecar config doesn't have a real hostname. In order to give some telemetry info, make a synthetic hostname.
+				telemetryMetadata: util.TelemetryMetadata{
```

Comment:
Maybe the naming is confusing here, but this actually configures stat_prefix and trace operator, not "Istio Telemetry".

My intent was to make it clear that it was *not* for critical config, and should not be used as such, since all of it is best effort (can have conflicts)

Reply:
Makes sense.